common/src/api/internal/nexus.rs

```diff
@@ -113,12 +139,44 @@ pub struct UpdateArtifactId {
     Ord,
     PartialOrd,
     Display,
+    FromStr,
     Deserialize,
     Serialize,
     JsonSchema,
+    EnumIter,
 )]
-#[display(style = "kebab-case")]
-#[serde(rename_all = "kebab-case")]
+#[display(style = "snake_case")]
+#[serde(rename_all = "snake_case")]
 pub enum UpdateArtifactKind {
-    Zone,
+    // Sled Artifacts
+    GimletSp,
+    GimletRot,
+    Host,
+    Trampoline,
+    ControlPlane,
+
+    // PSC Artifacts
+    PscSp,
+    PscRot,
+
+    // Switch Artifacts
+    SwitchSp,
+    SwitchRot,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use strum::IntoEnumIterator;
+
+    #[test]
+    fn update_artifact_kind_roundtrip() {
+        for kind in UpdateArtifactKind::iter() {
+            let as_string = kind.to_string();
+            let kind2 = as_string.parse::<UpdateArtifactKind>().unwrap_or_else(
+                |error| panic!("kind {as_string} parsed successfully: {error}"),
```

Comment:
Should this be "parsed unsuccessfully" or maybe "failed to parse"?

Reply:
Ah yeah! I'll fix it up in a followup (I'm thinking of redoing this code a bit anyway.)

Reply:
Fixed in https://github.com/oxidecomputer/omicron/pull/2369.